Stratis.Bitcoin.Features.BlockStore/LoopSteps/ProcessPendingStorageStep.cs

```diff
@@ -189,12 +214,39 @@ internal ProcessPendingStorageContext(BlockStoreLoop blockStoreLoop, ChainedBloc
         internal ConcurrentStack<BlockPair> PendingBlockPairsToStore = new ConcurrentStack<BlockPair>();
 
         /// <summary>
-        /// Gets the next block to process.
+        /// Break execution if:
+        /// <list>
+        ///     <item>1: Next block is null.</item>
+        ///     <item>2: Next block previous hash does not match previous block.</item>
+        ///     <item>3: Netx block is at tip.</item>
+        ///     <item>4: Store tip is as consensus.</item>
+        /// </list>
         /// </summary>
-        internal void GetNextBlock()
+        /// <returns>Returns <c>true</c> if none of the above condition were met, i.e. the next block can be processed.</returns>
+        internal bool CanProcessNextBlock()
         {
             this.InputChainedBlock = this.NextChainedBlock;
             this.NextChainedBlock = this.BlockStoreLoop.Chain.GetBlock(this.NextChainedBlock.Height + 1);
+
+            if (this.NextChainedBlock == null)
+                return false;
+            if (this.NextChainedBlock.Header.HashPrevBlock != this.InputChainedBlock.HashBlock)
+                return false;
+            if (this.NextChainedBlock.Height > this.BlockStoreLoop.ChainState.HighestValidatedPoW?.Height)
+                return false;
+            if (this.BlockStoreLoop.StoreTip.Height >= this.BlockStoreLoop.ChainState.HighestValidatedPoW?.Height)
```

Comment:
Do we need this check?
We know that InputChainedBlock is equal to StoreTip right?

Reply:
You are correct, I have removed this check.